backend/app/api/endpoints/investor.py

```
@@ -1,19 +1,70 @@
-async def delete_investor():
-    return {"message": "Investor deleted"}
 
-@router.get("/:details")
-async def get_investor_details_by_id():
-    return {"message": "Investor details by ID"}
+@router.get("/", response_model=List[InvestorWithUserRead])
+async def list_investors(
+    session: SessionDep,
+    skip: int = 0,
+    limit: int = 100,
+    current_user: uuid.UUID = Depends(get_current_user),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)